fdbserver/SimKmsConnector.actor.cpp

```diff
@@ -90,8 +108,8 @@ ACTOR Future<Void> ekLookupByIds(Reference<SimKmsConnectorContext> ctx,
 	for (const auto& item : req.encryptKeyInfos) {
 		const auto& itr = ctx->simEncryptKeyStore.find(item.baseCipherId);
 		if (itr != ctx->simEncryptKeyStore.end()) {
-			rep.cipherKeyDetails.emplace_back(
-			    item.domainId, itr->first, StringRef(rep.arena, itr->second.get()->key), rep.arena);
+			rep.cipherKeyDetails.emplace_back_deep(
+			    req.arena, item.domainId, itr->first, StringRef(itr->second.get()->key));
```

Comment:
This should be rep.arena instead of req.arena, right?

Reply:
Good catch! Yes it needs to be `reply` arena versus `request` arena. 
I think reason Valgrind didn't trip is because for test, both request and reply was in-scope during the usage of the method..